pkg/manifests/types.go

```diff
@@ -146,6 +146,9 @@ type DedicatedServiceMonitors struct {
 // The `KubeStateMetricsConfig` resource defines settings for the
 // `kube-state-metrics` agent.
 type KubeStateMetricsConfig struct {
+	// Comma-separated list of metrics not to be enabled.
+	// This list comprises exact metric names and/or regex patterns.
+	MetricDenylist []string `json:"metricDenylist,omitempty"`
```

Comment:
Since this will end up in our official docs and the behavior can be somewhat surprising, I think this doc string should be a bit more extensive.